src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -106,9 +106,15 @@ def get_discovered_devices(self, nodenames=None):
     def get_nfs_conf_url(self, nfs_cluster, instance):
         #
         # Fetch cephnfs object for "nfs_cluster" and then return a rados://
-        # URL for the instance within that cluster.
+        # URL for the instance within that cluster. If the fetch fails, just
+        # return None.
         #
-        ceph_nfs = self.rook_api_get("cephnfses/{0}".format(nfs_cluster))
+        try:
+            ceph_nfs = self.rook_api_get("cephnfses/{0}".format(nfs_cluster))
+        except ApiException as e:
+            log.info("Unable to fetch cephnfs object: {}".format(e.status))
+            return None
```

Comment:
Not a big fan of returning `None` in general. Seems to be ok  here though.

Reply:
I could change it to some arbitrary string (e.g. "Undefined"), but I worry other modules calling into this code and getting back some non-URL string and not being able to tell.